Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -23,6 +23,7 @@ ComponentInfo::ComponentInfo(
     boost::shared_ptr<const std::vector<size_t>> parentIndices,
     boost::shared_ptr<std::vector<Eigen::Vector3d>> positions,
     boost::shared_ptr<std::vector<Eigen::Quaterniond>> rotations,
+    boost::shared_ptr<std::vector<Eigen::Vector3d>> scaleFactors,
```

Comment:
It's actually quite well imbedded in may of the calculations of `ObjComponent`. But you're right. Only one algorithm to my knowledge actually writes these scale factors.